src/backend.rs

```diff
@@ -1,18 +1,23 @@
 use std::path::{Path, PathBuf};
 use std::collections::HashMap;
-use eyre::{Result, anyhow};
+use anyhow::{Result, anyhow};
 use tokio::fs;
 use tracing::info;
 use xdg::BaseDirectories;
-use veilid_core::{VeilidAPI, CryptoKey, VeilidUpdate, VeilidConfigInner, api_startup_config, DHTSchema, CRYPTO_KIND_VLD0, vld0_generate_keypair, CryptoTyped, CryptoSystemVLD0, RoutingContext, KeyPair, ProtectedStore}; // Added ProtectedStore here
+use veilid_core::{VeilidAPI, CryptoKey, VeilidUpdate, VeilidConfigInner, api_startup_config, DHTSchema, CRYPTO_KIND_VLD0, vld0_generate_keypair, CryptoTyped, CryptoSystemVLD0, RoutingContext, KeyPair, ProtectedStore, SharedSecret, CryptoSystem};
 use std::sync::Arc;
-use crate::group::{Group, GroupKeypair};
+use crate::common::{CommonKeypair, DHTEntity};
+use crate::group::Group;
+use crate::repo::Repo;
 
 pub struct Backend {
     path: PathBuf,
     port: u16,
-    veilid_api: Option<VeilidAPI>,
+    veilid_api: Option<Arc<VeilidAPI>>,
```

Comment:
Does this have to be an Arc? Maybe we could use a Box instead?

Reply:
If I switch to a Box, we find ourselves faced with a lack of persistence after shutdown.
It gets back to the initial issue where with the Box, the shutdown is successful but I can not retrieve the group name after shutdown

Reply:
I think that's okay. shutdown should be the last thing called and afterwards you should need to start veilid again to get a group and call get name on it.

Reply:
We can have the group id cloned to reference the group in between shutdowns

Reply:
If the table store isn't persisting group info, or the dht isn't persisting data we published, then we've got more of a problem.